src/main/java/lotto/controller/LottoController.java

```diff
@@ -27,10 +25,35 @@ public LottoController() {
 
     public void buyLotto() {
         money = new Money(lottoUI.getMoneyFromUser());
-        lottos = Lottos.fromMoney(money, lottoNumberGenerator);
+        NumOfLotto numOfManual = new NumOfLotto(lottoUI.getNumOfManualFromUser());
+        NumOfLotto numOfAuto = money.howMany().minus(numOfManual);
+
+        List<Lotto> manualLottoList = buyManualLotto(numOfManual);
+        List<Lotto> autoLottoList = buyAutoLotto(numOfAuto, lottoNumberGenerator);
+
+        lottoUI.printLottosMessage(numOfManual, numOfAuto);
+        lottos = Lottos.of(manualLottoList, autoLottoList);
         lottoUI.printLottos(LottosDto.from(lottos));
     }
 
+    private List<Lotto> buyManualLotto(NumOfLotto numOfManual) {
+        lottoUI.printManualLottoMessage();
+        List<Lotto> lottoList = new ArrayList<>();
+        for (int i = 0; i < numOfManual.getNumOfLotto(); i++) {
+            lottoList.add(Lotto.manualGenerate(
+                    lottoUI.getManualLottoFromUser()));
+        }
+        return lottoList;
+    }
+
+    private List<Lotto> buyAutoLotto(NumOfLotto numOfAuto, LottoNumberGenerator lottoNumberGenerator) {
+        List<Lotto> lottoList = new ArrayList<>();
+        for (int i = 0; i < numOfAuto.getNumOfLotto(); i++) {
+            lottoList.add(Lotto.autoGenerate(lottoNumberGenerator));
```

Comment:
```suggestion
            lottoList.add(new Lotto(lottoNumberGenerator.generate()));
```
또는 
```suggestion
            lottoList.add(Lotto.of(lottoNumberGenerator.generate()));
```

정도로만 하면 Lotto 내부의 로직도 정리되고 좀더 간결해지지 않을까요?

Reply:
생각해보니 불필요한 메서드 같네요..! ManualNumberGenerator 관련 답변주신 내용이랑 같이 고민해서 개선해보겠습니다:) 감사합니다!!